bpf/proxy/syncer.go

```diff
@@ -221,9 +221,62 @@ func (s *Syncer) loadOrigs() error {
 	return nil
 }
 
-func (s *Syncer) startupSync(state DPSyncerState) error {
+type syncRef struct {
+	svc  k8sp.ServicePortName
+	info k8sp.ServicePort
+}
+
+func (s *Syncer) svcMapToIPPortProtoMap(svcs k8sp.ServiceMap) map[nat.FrontendKey]syncRef {
+	ref := make(map[nat.FrontendKey]syncRef, len(svcs))
+
+	for key, svc := range svcs {
+		clusterIP := svc.ClusterIP()
+		proto := uint8(ProtoV1ToIntPanic(svc.Protocol()))
+		port := uint16(svc.Port())
+
+		xref := syncRef{key, svc}
+
+		ref[nat.NewNATKey(clusterIP, port, proto)] = xref
+
+		np := uint16(0)
+
+		if svc.NodePort() != 0 {
+			np = uint16(svc.NodePort())
+
+			ref[nat.NewNATKey(clusterIP, np, proto)] = xref
+
+			for _, npIP := range s.nodePortIPs {
+				ref[nat.NewNATKey(npIP, np, proto)] = xref
+			}
+		}
+
+		for _, extIP := range svc.ExternalIPStrings() {
+			ref[nat.NewNATKey(net.ParseIP(extIP), port, proto)] = xref
+
+			// FIXME we do not create these, not sure if we should create np for
+			// external ip and so consider it here as well
+			//
+			/*
+				if np != 0 {
+					ref[NewNATKey(extIP, np, svc.Protocol())] = xref
+				}
+			*/
+		}
+	}
+
+	return ref
+}
+
+func (s *Syncer) startupBuildPrev(state DPSyncerState) error {
```

Comment:
Worth explaining the big idea behind this sync somewhere.  I think the main goal is to figure out the IDs that were used in the dataplane previously in order to minimise dataplane churn/inconsistency when doing the initial update.